.circleci/config.yml

```diff
@@ -66,5 +78,31 @@ workflows:
               only:
                 - master
                 - /\d+\.\d+\.x/
+  build_deploy_releases:
+    jobs:
+      - build:
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v\d+\.\d+\.\d+(-rc\.\d+)?$/
+      - deploy:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v\d+\.\d+\.\d+(-rc\.\d+)?$/
+      - staging-approval:
+          type: approval
```

Comment:
TODO: Test what this approval job looks like in CircleCI.

Reply:
Here are some screenshots from the deploy pipeline I'm working on for the docs site:

<img width="503" alt="image" src="https://user-images.githubusercontent.com/8924140/54522803-70adfb00-49b1-11e9-826c-c879e12d8457.png">

<img width="977" alt="image" src="https://user-images.githubusercontent.com/8924140/54522827-815e7100-49b1-11e9-93ed-e599c7cc19c6.png">

When you click on the approval type job, you get a pop-up like the following:
<img width="544" alt="image" src="https://user-images.githubusercontent.com/8924140/54522847-8cb19c80-49b1-11e9-99ef-e712fd57e8ed.png">

Reply:
The official docs don't specify who has access to approve `approval` type jobs, but I did ask @adriancole as a test to try approving it, which seemed to do nothing. I am able to approve and cause the workflow to proceed so maybe it is limited to users on the CircleCI team for the GitHub org?
https://circleci.com/docs/2.0/workflows/#holding-a-workflow-for-a-manual-approval

Reply:
see https://circleci.com/workflow-run/d0c4004c-0334-4237-83ac-1f5f385d881c